llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -10154,7 +10154,10 @@ SDValue RISCVTargetLowering::lowerVectorMaskVecReduction(SDValue Op,
   case ISD::VP_REDUCE_AND: {
     // vcpop ~x == 0
     SDValue TrueMask = DAG.getNode(RISCVISD::VMSET_VL, DL, ContainerVT, VL);
-    Vec = DAG.getNode(RISCVISD::VMXOR_VL, DL, ContainerVT, Vec, TrueMask, VL);
+    if (IsVP || VecVT.isFixedLengthVector())
```

Comment:
Do we need the special case?  I'm somewhat surprised that the plain ISD::XOR doesn't work out fine for the fixed case at least.  Any idea why?  Even for the VP case, I think we only need the mask and VL on the vcpop itself?

Reply:
It caused an extra vtype toggle for fixed vectors. I didn't dig any further.